R/PqConnection.R

```diff
@@ -308,6 +308,19 @@ postgresWaitForNotify <- function (conn, timeout = 1) {
   if ('pid' %in% names(out)) out else NULL
 }
 
+#' Return whether a transaction is ongoing
+#'
+#' Detect whether the transaction is active for the given connection. A
+#' transaction might be started with [dbBegin()] or wrapped within
+#' [DBI::dbWithTransaction()].
+#' @export
+#' @param conn a [PqConnection-class] object, produced by
+#'   [DBI::dbConnect()]
+#' @return A boolean, indicating if a transaction is ongoing.
+postgresIsTransacting <- function(conn) {
+  connection_is_transacting(conn)
```

Comment:
This should be a pointer.
```suggestion
  connection_is_transacting(conn@ptr)
```